### PR TITLE
feature: add search to map component

### LINKED ISSUE #299: Landscape creation: add leaflet map with pin drop tool
* when loading page for initial creattion, center map on country
* when saving, save both lat/lng of pin and zoom level

image

### PR DESCRIPTION
**Pull request by paulschreiber**
## Description
Add a search box to map component. When a location is found, a draggable pin is inserted.

### Checklist
- [X] Corresponding issue has been opened
- [ ] New tests added
- [X] Verified on mobile
- [X] Verified on desktop

### Related Issues
Fixes #299.

### CODE REVIEW

File: src/gis/components/Map.js [hunk 1]
```diff
@@ -1,9 +1,79 @@
...
 import { v4 as uuidv4 } from 'uuid';
+import { GeoSearchControl, OpenStreetMapProvider } from 'leaflet-geosearch';
+import L from 'leaflet';
+
+import 'leaflet/dist/leaflet.css';
+import 'leaflet-geosearch/dist/geosearch.css';
 
 import 'gis/components/Map.css';
 
+delete L.Icon.Default.prototype._getIconUrl;
```
Comment (josebui) [inline src/gis/components/Map.js:12]: Question: Why is this needed?

**Review by josebui — APPROVED** [review]
Looks good, just one question

File: src/gis/components/Map.js (hunk 1 above)
Comment (paulschreiber) [reply src/gis/components/Map.js:12]: If we don't include it, the marker icon is broken.
